### PR TITLE
Verify with NSEC3

### PR DESCRIPTION
**Pull request by khibino**
Add logic to prove empty RRset or non-existence of RRset with NSEC3 records.

This logic recognizes the following cases where NSEC3 records would be returned:

* Name Error - domain name does not exist
* No Data - domain name exists, but RRset is empty for specified type
* Delegate not signed domain name with OptOut flag
* Wildcard Expansion - domain name matches wildcard, but not wildcard name does not exist
* Wildcard No Data - domain name matches wildcard, but RRest is empty for specified type

### CODE REVIEW

File: dnsext-dnssec/dnsext-dnssec.cabal [hunk 1]
```diff
@@ -45,6 +47,7 @@ library
         dnsext-types,
         bytestring,
         containers,
+        base32,
```
Comment (kazu-yamamoto) [inline dnsext-dnssec/dnsext-dnssec.cabal:50]: With GHC 9.4.3, `base32` (which depends on text) cannot be resolved.
Would you try GHC 9.4.3 by yourself?
One alternative is using `base32-bytestring`, I guess.

**Review by kazu-yamamoto — CHANGES_REQUESTED** [review]
Most ports looks good.
But I cannot build this PR.

File: dnsext-dnssec/dnsext-dnssec.cabal (hunk 1 above)
Comment (kazu-yamamoto) [reply dnsext-dnssec/dnsext-dnssec.cabal:50]: Depending `text` is a bad idea, anyway.

Comment (khibino) [reply dnsext-dnssec/dnsext-dnssec.cabal:50]: `base32-bytestring` has build errors

```
Building library for base32-bytestring-0.2.1.0..
[1 of 3] Compiling Data.ByteString.Base32.Internal ( src/Data/ByteString/Base32/Internal.hs, dist/build/Data/ByteString/Base32/Internal.o, dist/build/Data/ByteString/Base32/Internal.dyn_o )

src/Data/ByteString/Base32/Internal.hs:73:1: error:
    Parse error in pattern: unpack5Ptr In a function binding for the ‘@’ operator.
                            Perhaps you meant an as-pattern, which must not be surrounded by whitespace
   |
73 | unpack5Ptr !tbl bs @ (PS fptr off sz) =
   | ^^^^^^^^^^^^^^^^^^
```

Comment (khibino) [reply dnsext-dnssec/dnsext-dnssec.cabal:50]: I found the encoder in dnsext-types/DNS/Types/Base32Hex.hs.
What do you think about adding a decoder there?

Comment (kazu-yamamoto) [reply dnsext-dnssec/dnsext-dnssec.cabal:50]: Go ahead.

Comment (khibino) [reply dnsext-dnssec/dnsext-dnssec.cabal:50]: After merging #50, rebase this PR and add the fixes

**Pull request closed without merging**